shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -1538,4 +1541,64 @@ void updateSemantics(
   public interface AsyncWaitForVsyncDelegate {
     void asyncWaitForVsync(final long cookie);
   }
+
+  // ----- Start Frame Time Listener Support -----
+
+  /**
+   * Interface providing a callback for Flutter's engine to notifies Android of a frame time event.
+   *
+   * <p>This callback may be invoked from any thread. Implementers of {@link OnFrameTimeListener}
+   * should ensure to check and handle the current thread appropriately, especially if subsequent
+   * actions are expected to run on the main thread.
+   */
+  public interface OnFrameTimeListener {
+
+    /**
+     * Callback triggered when the rasterizer is beginning its rendering process.
+     *
+     * @param buildStartTime Timestamp (in nanoseconds) when build process started.
+     * @param buildEndTime Timestamp (in nanoseconds) when build process completed.
+     * @param rasterStartTime Timestamp (in nanoseconds) when the rasterizer begin its rendering.
+     * @param currentNanoTime The value of System.nanoTime() when the rasterizer begin its
+     *     rendering.
+     */
+    default void onRasterStart(
+        long buildStartTime, long buildEndTime, long rasterStartTime, long currentNanoTime) {}
```

Comment:
Because Image.getTimestamp() returns nano time, and build end time which is sent from dart side and engine side are millisecondsSinceEpoch. I explained the mechanism in the ticket [133990](https://github.com/flutter/flutter/issues/133990). Actually, I only need long buildEndTime and long currentNanoTime to implement it, the other parameters are reserved for future use.

Reply:
And the currentNanoTime is from another source (System.nanoTime()). The others are nanoseconds since epoch.

Reply:
I removed currentNanoTime since the plugin can acquire it using System.nanoTime()